applications/CoSimulationApplication/python_scripts/data_transfer_operators/copy_one_to_many.py

```diff
@@ -0,0 +1,61 @@
+# Importing the base class
+from KratosMultiphysics.CoSimulationApplication.base_classes.co_simulation_data_transfer_operator import CoSimulationDataTransferOperator
+
+def Create(settings):
+    return CopyOneToMany(settings)
+
+class CopyOneToMany(CoSimulationDataTransferOperator):
+    """DataTransferOperator to take one single value and set it to all values on another interface.
+    Used e.g. for FSI with SDof, where the SDof has one value and the fluid interface has many
+    """
+    def _ExecuteTransferData(self, from_solver_data, to_solver_data, transfer_options):
+        if not to_solver_data.IsDefinedOnThisRank():
+            return
+
+        data_value = 0.0
+        rank_for_broadcast = 0
+        if from_solver_data.IsDefinedOnThisRank():
+            data_value = from_solver_data.GetData()
```

Comment:
data_value overwrites the predefined data_value = 0.0 without being sure to get a new value in L.20

Reply:
yes only one rank has the actual value => check in line 19

Reply:
But for a rank with 0 nodes, the variable is still defined on this rank. But the data value size is 0. So it tries to Broadcast in L.24 the data_value from L.18 which is an empty list.

Reply:
yes that is correct
I am a bit confused, does it give an error?
the previous code used the same principle from what I can see :thinking:

Reply:
now it should be correct right?